sidecar/src/config.rs

```diff
@@ -182,6 +182,7 @@ pub fn get_product_endpoint(subdomain: &str, endpoint: &Endpoint) -> Endpoint {
         Endpoint {
             url: hyper::Uri::from_parts(parts).unwrap(),
             api_key: Some(api_key.clone()),
+            ..Default::default()
```

Comment:
This also applies to the data-pipeline implementation...Is the current default of 10s appropriate? And do we want the same timeout value for both agent and agentless communication? I would think we would want a much lower value for talking to a trace agent? I'm not sure what appropriate values are, maybe we could check with trace agent folks?

Reply:
This code should inherit the endpoint from input endpoint var.

Reply:
I think it's the responsibility of the code setting the agentless mode to also decide on the timeout. But yes, I have no idea what proper timeouts are. Probably `Endpoint::from_url()` may default to a lower 3 sec timeout?